Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -642,6 +657,10 @@ double MatrixWorkspace::getXMax() const {
 }
 
 void MatrixWorkspace::getXMinMax(double &xmin, double &xmax) const {
+  if (m_indexInfo->communicator().size() > 1)
+    throw std::runtime_error(
+        "MatrixWorkspace: Parallel support for XMin and XMax not implemented.");
```

Comment:
Would not be hard to do this, but for the time being I am not doing this. The main issue here is that this cannot be supported unless a workspace always has a `Parallel::Communicator`. Currently `MatrixWorkspace` does, via `IndexInfo`, so this would be fine but maybe this requires more thought?